src/conn/conn_api.c

```diff
@@ -2199,6 +2201,8 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
 	__wt_scr_free(session, &i2);
 	__wt_scr_free(session, &i3);
 
+	if (free_merge_cfg)
+		__wt_free(session, merge_cfg);
```

Comment:
Then I'd suggest:
```
diff --git a/src/conn/conn_api.c b/src/conn/conn_api.c
index 25db237..5d5244e 100644
--- a/src/conn/conn_api.c
+++ b/src/conn/conn_api.c
@@ -1887,7 +1887,7 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
        WT_DECL_RET;
        const WT_NAME_FLAG *ft;
        WT_SESSION_IMPL *session;
-       bool config_base_set, free_merge_cfg;
+       bool config_base_set;
        const char *enc_cfg[] = { NULL, NULL }, *merge_cfg;
        char version[64];
 
@@ -1899,7 +1899,7 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
 
        conn = NULL;
        session = NULL;
-       free_merge_cfg = false;
+       merge_cfg = NULL;
 
        WT_RET(__wt_library_init());
 
@@ -2025,10 +2025,11 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
                 * read-only are tested in their individual locations later.
                 */
                __conn_config_readonly(cfg);
-               free_merge_cfg = true;
                WT_ERR(__wt_config_merge(session, cfg, NULL, &conn->cfg));
-       } else
+       } else {
                conn->cfg = merge_cfg;
+               merge_cfg = NULL;
+       }
 
        /*
         * Configuration ...
@@ -2201,8 +2202,8 @@ err:      /* Discard the scratch buffers. */
        __wt_scr_free(session, &i2);
        __wt_scr_free(session, &i3);
 
-       if (free_merge_cfg)
-               __wt_free(session, merge_cfg);
+       __wt_free(session, merge_cfg);
+
        /*
         * We may have allocated scratch memory when using the dummy session or
         * the subsequently created real session, and we don't want to tie down
```

Basically, to "give away" the memory, clear the original, and there's never more than a single reference.

I don't like it when two variables have to stay in sync, it always makes me nervous that code movement in the future will break things.